libflate_lz77/Cargo.toml

```
@@ -16,11 +16,12 @@ coveralls = {repository = "sile/libflate"}
 
 [dependencies]
 rle-decode-fast = "1.0.0"
-core2 = { version = "0.4", default-features = false, features = ["alloc"], optional = true }
-hashbrown = { version = "0.13", optional = true }
+core2 = { version = "0.4", default-features = false, features = ["alloc"] }
+hashbrown = { version = "0.13" }
 
 [dev-dependencies]
-libflate = { path = "../", version = "1" }
+libflate = { path = "../", version = "1", default-features = false }
```

```suggestion
libflate = { path = "../", version = "2", default-features = false }
```

Or, just remove the version completely. Since we're using a path dependency here, the version isn't necessary.

---

IIUC it's necessary for crates.io and docs.rs to have version declarations in order to build correctly

---

That's usually correct, but with path dependencies you don't need to specify version. You *can* specify the version, but if you do, it has to match the `Cargo.toml` version (meaning you would need to make the above edit).

---

I will be keeping both `version` and `path` specifiers.

> with `path` dependencies you don't need to specify version

The issue is with building tests using crates.io sources, since `cargo publish` packages the source of just that crate as a tarball, instead of having access to the full git repository. So, if a dependency is specified with just a `path` specifier, I doubt docs.rs or crates.io-side unit tests will be able to run correctly. Cargo calls this out [in its documentation](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations): 

> The `git` or `path` dependency will be used locally (in which case the `version` is checked against the local copy), and when published to a registry like [crates.io](https://crates.io/), it will use the registry version.

For an example in the wild, see `serde`, which as a `dev-dependency` imports `serde_derive` using both `path` and `version` specifiers.

---

> For an example in the wild, see [serde](https://github.com/serde-rs/serde/blob/107018c6285e87f4cb265e1ef6c3d7931c1a15eb/serde/Cargo.toml#L21), which in its dev-dependencies imports serde_derive using both path and version specifiers.

Right on, then the suggested edit should work. Necessary with the major version bump.

---

Done. I went with the same style as writing `2.0` instead of `2` if that makes a difference to you.